lib/instrument/headers.js

```diff
@@ -0,0 +1,260 @@
+import {archive, unarchive} from './transformer/nskeyed';
+import plistlib from 'bplist-parser';
+
+const FLAGS_TYPE = 2;
+const DTX_MESSAGE_HEADER_LENGTH = 32;
+const DTX_MESSAGE_PAYLOAD_HEADER_LENGTH = 16;
+const DTX_MESSAGE_HEADER_MAGIC = 0x1F3D5B79;
+
+class DTXMessageHeader {
+  static build (data) {
+    const messageHeader = Buffer.alloc(DTX_MESSAGE_HEADER_LENGTH);
+    messageHeader.writeUInt32LE(DTX_MESSAGE_HEADER_MAGIC, 0);
+    messageHeader.writeUInt32LE(DTX_MESSAGE_HEADER_LENGTH, 4);
+    messageHeader.writeUInt16LE(0, 8);
+    messageHeader.writeUInt16LE(1, 10);
+    messageHeader.writeUInt32LE(data.payloadLength, 12);
+    messageHeader.writeUInt32LE(data.identifier, 16);
+    messageHeader.writeUInt32LE(data.conversationIndex, 20);
+    messageHeader.writeUInt32LE(data.channel, 24);
+    messageHeader.writeUInt32LE(data.expectsReply, 28);
+    return messageHeader;
+  }
+  static parse (headerBuffer) {
+    return {
+      magic: headerBuffer.readUInt32LE(0),
+      headerLength: headerBuffer.readUInt32LE(4),
+      fragmentId: headerBuffer.readUInt16LE(8),
+      fragmentCount: headerBuffer.readUInt16LE(10),
+      payloadLength: headerBuffer.readUInt32LE(12),
+      identifier: headerBuffer.readUInt32LE(16),
+      conversationIndex: headerBuffer.readUInt32LE(20),
+      channel: headerBuffer.readUInt32LE(24),
+      expectsReply: headerBuffer.readUInt32LE(28),
+    };
+  }
+}
+
+class DTXMessagePayloadHeader {
+  static build (data) {
+    const messageHeader = Buffer.alloc(DTX_MESSAGE_PAYLOAD_HEADER_LENGTH);
+    messageHeader.writeUInt32LE(data.flags, 0);
+    messageHeader.writeUInt32LE(data.auxLength, 4);
+    // eslint-disable-next-line no-undef
+    messageHeader.writeBigUInt64LE(BigInt(data.totalLength), 8);
```

Comment:
should we explicitly cast this value? I would expect data.totalLength is already of the expected type because if it is not then this conversion does not make much sense - the value would be already cut to 52 bits